### PR TITLE
Showing new submissions

### PR DESCRIPTION
**Pull request by avdata99**
Related to #679 

 - Show new submissions available in the survey list
 - Allow updating resources from survey list and the dataset page

## Survey list
New submissions detected

Selection_353

Job started

Selection_354

Running job detected
Selection_352

## Dataset level

nosub

sub

### CODE REVIEW

**Comment by amercader** [comment]
@avdata99 I had a play with this. I don't know if it was because I tried to refresh an existing Kobo dataset but the process didn't work as expected for me
Screenshot 2021-09-01 at 12-35-47 KoBo - RIDL
When clicking "Update resource" a job was triggered and logs showed that it finished ok. But when refreshing the page the "Update resource" button reappeared and all the resources were deleted.

I tried to delete and recreate the dataset but now I can't try it again because it doesn't detect new submissions. Can you give me access to other surveys, or describe how I can create my own on Kobo?

Some more comments:

On the dataset page can we combine the "Check for updates" and "Update resources" step into a single one? Just "Update resources" and check if there are new submissions directly, and return a flash messages if not.
In fact, let's just remove this yellow box and add a "Refresh KoBo data" button in the top right corner, between "Copy" and "Manage"
Screenshot 2021-09-01 at 14-01-37 Data in IDP situations - RIDL

I'll follow up with more comments after the code review

**Comment by avdata99** [comment]
@amercader old version of KoBo resources do not include the submission counter, that's why `Data in IDP` shows all submission as new.
Maybe you need to clean you DB before testing.
If you start from scratch with `My third Survey` you can easily add new submissions (I shared privately the access to this survey)

**Comment by avdata99** [comment]
New UI @amercader 

nosub

sub

Changed [here](https://github.com/okfn/ckanext-unhcr/commit/e5e3032ade920a05561ee631e07095c7a5ceb3f6)

**Comment by amercader** [comment]
Thanks @avdata99, sorry I wasn't clear. I don't think we need two AJAX calls, one for checking if there are submissions and one for updating the resources. We just do one call to the update endpoint, and in the server we check if there are new submissions. If not the response is "No new submissions" and if yes, "Update job started" Does this make sense?
I think this will simplify things

**Comment by amercader** [comment]
One thing I've noticed is that we are starting an awful lot of background jobs when updating the resources. I added a small change to the core jobs module to know what exactly was being called and not surprisingly, apart from the expected to `update_pkg_kobo_resources` and `download_kobo_export` it was mostly calls to `process_dataset_on_update` (about a 12 of them). This get triggered on each resource/package update/patch, probably one per resources in the dataset. These are used to change values of the dataset metadata but we really don't need to do it every time we do a Kobo import. There is a context key that avoids triggering the `process_dataset_on_update` jobs:

```diff
diff --git a/ckanext/unhcr/kobo/kobo_dataset.py b/ckanext/unhcr/kobo/kobo_dataset.py
index a660dc0..29e1714 100644
--- a/ckanext/unhcr/kobo/kobo_dataset.py
+++ b/ckanext/unhcr/kobo/kobo_dataset.py
@@ -251,7 +251,7 @@ class KoboDataset:
     def update_resource(self, resource_dict, local_file, user_name, submission_count):
         """ Update the resource with real data """
 
-        context = {'user': user_name}
+        context = {'user': user_name, 'job': True}
         if not resource_dict:
             raise toolkit.ValidationError({'resource': ["empty resource to update"]})
         kobo_details = resource_dict.get('kobo_details')
@@ -329,7 +329,7 @@ class KoboDataset:
         }
 
         resource = toolkit.get_action('resource_patch')(
-            {'user': user_obj.name},
+            {'user': user_obj.name, 'job': True},
             patch_resource,
         )
 
@@ -384,7 +384,7 @@ class KoboDataset:
             patch_resource['kobo_details']['kobo_download_attempts'] = 0
 
             toolkit.get_action('resource_patch')(
-                {'user': user_obj.name},
+                {'user': user_obj.name, 'job': True},
                 patch_resource,
             )

```

Can you try to apply this and do some testing to see if it affects the other fields of the dataset? Specially the resources visibility

**Comment by avdata99** [comment]
> Thanks @avdata99, sorry I wasn't clear. I don't think we need two AJAX calls, one for checking if there are submissions and one for updating the resources. We just do one call to the update endpoint, and in the server we check if there are new submissions. If not the response is "No new submissions" and if yes, "Update job started" Does this make sense?
> I think this will simplify things

@amercader Moved to a sigle call [here](https://github.com/okfn/ckanext-unhcr/commit/4a336da7f63c830a6e95455c3094ab5e2c1d3170) (also fixed UI if the user re-try update)

File: ckanext/unhcr/kobo/kobo_dataset.py [hunk 1]
```diff
@@ -250,3 +274,119 @@ def update_resource(self, resource_dict, local_file, user_name):
...
+            else:
+                export_id = ''
+
+            # prepare the update
+            kobo_details = resource['kobo_details']
+            patch_resource = {
+                'id': resource['id'],
+                'date_range_start': date_range_start,
+                'date_range_end': date_range_end,
+                'kobo_details': kobo_details
```
[the remark this answers is not in the document]
Comment (avdata99) [reply ckanext/unhcr/kobo/kobo_dataset.py:383]: Added and showed to users [here](https://github.com/okfn/ckanext-unhcr/commit/be7995c5c8979185987a29d79106842d39806d58)

**Review by amercader — APPROVED** [review]

**Comment by amercader** [comment]
@avdata99 I pushed some tweaks 253125a and marked this for approval. I believe there are a couple of things we can iron out but we can do so on separate PRs

* Avoid datapusher submits. Datapusher has an `IDataPusher` [interface](https://github.com/ckan/ckan/blob/c1ba2dbd235ea8ce2c532a08f39392a79adccb6f/ckanext/datapusher/interfaces.py#L15) with a `can_upload()` method that maybe we can use for this. It's a bit rubbish because it only gives you the resource id but maybe there's something in the metadata that you can use to skip the submission
* I think the resources Kobo last_update should be more prominent to inform publishers and users. I'll probably make it a dataset level field. It doesn't need to be a schema field, just a virtual field that we set up on the dataset at index time (`IPackageController.before_index()`), getting it from one of the Kobo resources. I'd then display it in the "Data Files" section of the dataset page (and on the Surveys table list if possible). Again, this can be done later on.

Overall looks great though, and I think it's ready to be deployed to UAT

**Pull request merged**